MDNetworking/Wrappers/MDList.cs

```diff
@@ -587,152 +671,172 @@ public void DoFullResynch(List<T> NewList)
 
         public ReadOnlyCollection<T> AsReadOnly()
         {
-            return RealList.AsReadOnly();
+            return AsList().AsReadOnly();
+        }
+
+        protected List<T> AsList()
+        {
+            List<T> ReadOnlyList = new List<T>();
+            RealList.ForEach((KeyValuePair<T, IMDDataConverter> Pair) => ReadOnlyList.Add(Pair.Key));
+            return ReadOnlyList;
         }
         
         public int BinarySearch(int index, int count, T item, IComparer<T> comparer)
         {
-            return RealList.BinarySearch(index, count, item, comparer);
+            return RealList.BinarySearch(index, count, CreateNewItem(item), new CustomComparer(comparer));
         }
 
         public int BinarySearch(T item)
         {
-            return RealList.BinarySearch(item);
+            return RealList.BinarySearch(CreateNewItem(item), new CustomComparer(null));
         }
         
         public int BinarySearch(T item, IComparer<T> comparer)
         {
-            return RealList.BinarySearch(item, comparer);
+            return RealList.BinarySearch(CreateNewItem(item), new CustomComparer(comparer));
         }
         
         public bool Contains(T item)
         {
-            return RealList.Contains(item);
+            foreach (KeyValuePair<T, IMDDataConverter> listItem in RealList)
+            {
+                if (listItem.Key == null)
```

Comment:
`if (listItem.Key == null && item == null)`?

Reply:
No it is correct as is, it wouldnt work otherwise.
If item was not null and listItem.Key is null, then the else if would crash when it tries to do .Equals on a null. I could probably have made this more obvious